custom/icds_reports/utils/aggregation.py

```diff
@@ -931,7 +931,11 @@ def data_from_ucr_query(self):
                 FIRST_VALUE(form_date) OVER forms as first_submission,
                 LAST_VALUE(form_date) OVER forms as last_submission
             FROM "{ucr_tablename}"
-            WHERE inserted_at >= %(last_sync)s AND form_date <= %(now)s
+            WHERE awc_id IN (
+                SELECT DISTINCT awc_id
+                FROM "{ucr_tablename}"
+                WHERE inserted_at >= %(last_sync)s AND form_date <= %(now)s
+            )
```

Comment:
It looks like this will load the entire table into memory, instead of just the current days records. I think that ideally this would be solved with something like

```sql
UPDATE inactive_awws AS current_table SET
  first_submission = LEAST(current_table.first_submission, todays_values.first_submission)
FROM (query for today's values) todays_values
where current_table.awc_id = todays_values.awc_id
```